src/Schema/DataObject/InheritanceBuilder.php

```diff
@@ -52,12 +52,17 @@ public function fillAncestry(ModelType $modelType): void
         if (empty($ancestors)) {
             return;
         }
-        $parent = $ancestors[0];
-        $parentModel = $this->getSchema()->findOrMakeModel($parent);
+        foreach ($ancestors as $ancestor) {
+            $ancestorModel = $this->getSchema()->findOrMakeModel($ancestor);
+            if (!$ancestorModel || $ancestorModel->getName() === $modelType->getName()) {
+                continue;
+            }
+            break;
+        }
```

Comment:
```suggestion
        $ancestorModel = null;
        foreach ($ancestors as $ancestor) {
            $ancestorModel = $this->getSchema()->findOrMakeModel($ancestor);
            if ($ancestorModel && $ancestorModel->getName() !== $modelType->getName()) {
                break;
            }
            $ancestorModel = null;
        }
        if (!$ancestorModel) {
            return;
        }
```
Even if there's an ancestor in the `$ancestors` array, if we don't find an ancestor that _isn't_ identical (due to injection) to the current `$modelType`, we don't want do any of the rest of this stuff.

Also swapped the condition around, so instead of "break if we don't continue", we only break once we've found an ancestor. This is a more intuitive flow.

Reply:
Ah, I think while this seems seems like simple inverted logic to make it read better, I think this may introduce an error in some scenarios.  I'm not entirely sure so I'm inclined to not change it right now.

The 
```
if (!$ancestorModel) {
    return;
}
```
Should probably actually throw an exception as previously it would fail when it would go `$ancestorModel->getFieldByName`.  However you could also been in a weird situation $ancestorModel is null, though $modelTypes->getFields() is an empty array so you never actually get to that point.

I don't know, I'm just speculating for stuff that might happen

I think I'd prefer to just leave as and not risk causing additional regressions here.

Reply:
Lets talk about this offline - there's a lot going on here and there are definitely some changes required (the `if ($modelType->getName() !== $ancestorModel->getName()) {` condition at the bottom could result in some ancestry further up the chain not being processed, or at least _looks_ like it could which is arguably just as bad) - but we should work out whether the problem you've perceived with my recommendation is real or otherwise avoidable.

Reply:
The `findOrMakeModel()` method _cannot_ return null. It's got a concrete return type of `ModelType`. So before if we'd had a `if (!$parentModel)` condition, it simply would never have been true.

The scenario you're describing was impossible before, and will only be possible now because in my suggestion we _explicitly_ set the variable to null in scenarios where the model we have for that variable is not value for the logic that follows.

Reply:
> The scenario you're describing was impossible before, and will only be possible now because in my suggestion we explicitly set the variable to null in scenarios where the model we have for that variable is not value for the logic that follows.

I don't understand this sentence

Feels like we're dragging this out for the sake of making some code that works read a little nicer? This PR has a test showing things work as they should, can't we just merge this as is?

Reply:
That sentence basically says: You described a scenario that you think should throw an exception. The scenario where your `if (!$ancestorModel)` condition resolves to `true` wasn't possible before this PR, nor in this PR as it currently stands. The only reason it will potentially resolve to `true` _with_ my suggested change is because we _explicitly_ set it to null. The actual returned value from `findOrMakeModel()` will _never_ be null, so we don't need to worry about that. There was never a scenario where we would have gotten to `$ancestorModel->getFieldByName()` where `$ancestorModel` was null in your current PR, nor prior to this PR if we exchange `$ancestorModel` for `$parentModel`.

It is a response to "I don't know if any of this stuff would happen, I'm just speculating for stuff that might happen" - basically, none of that stuff _would_ nor _could_ happen.

> Feels like we're dragging this out for the sake of making some code that works read a little nicer?

Nope. I'm recommending this change in part because it increases code readability, in part because it reduces _confusion_, and in part because it reduces _complexity_. It is an all-round good suggestion that make the code easier to maintain and I can't see any reason to _not_ apply the change. Please apply the change.

Reply:
done